extensions/svelte/languages/svelte/injections.scm

```diff
@@ -1,74 +1,67 @@
-  (#set! "language" "javascript"))
+; Match script tags with a lang attribute
+(script_element
+    (start_tag
+        (attribute
+            (attribute_name) @_attr_name
+            (#eq? @_attr_name "lang")
+            (quoted_attribute_value
+                (attribute_value) @language
+            )
+        )
+    )
+    (raw_text) @content
+)
 
-; match javascript
-((script_element
-  (start_tag
-    (attribute
-      (attribute_name) @_name
-      (quoted_attribute_value (attribute_value) @_value)))
-    (raw_text) @content)
-  (#eq? @_name "lang")
-  (#eq? @_value "js")
-  (#set! "language" "javascript"))
+; Match script tags without a lang attribute
+(script_element
+    (start_tag
+        (attribute
+            (attribute_name) @_attr_name
+        )*
+    )
+    (raw_text) @content
+    (#not-any-of? @_attr_name "lang")
+    (#set! language "javascript")
+)
 
-; match typescript
-((script_element
-  (start_tag
-    (attribute
-      (attribute_name) @_name
-      (quoted_attribute_value (attribute_value) @_value)))
-    (raw_text) @content)
-  (#eq? @_name "lang")
-  (#eq? @_value "ts")
-  (#set! "language" "typescript"))
 
-(style_element
-  (raw_text) @content
-  (#set! "language" "css"))
-
-; match style tags without a lang tag
-((style_element
-  (start_tag
-    (attribute
-      (attribute_name) @_name)*)
-    (raw_text) @content)
-  (#not-eq? @_name "lang")
-  (#set! "language" "css"))
+; Mark everything as typescript because it's
+; a more generic superset of javascript
+; Not sure if it's possible to somehow refer to the
+; script's language attribute here.
+((svelte_raw_text) @content
+    (#set! "language" "ts")
+)
 
-; match css
-((style_element
-  (start_tag
-    (attribute
-      (attribute_name) @_name
-      (quoted_attribute_value (attribute_value) @_value)))
-    (raw_text) @content)
-  (#eq? @_name "lang")
-  (#eq? @_value "css")
-  (#set! "language" "css"))
+; Match style tags with a lang attribute
+(style_element
+    (start_tag
+        (attribute
+            (attribute_name) @_attr_name
+            (#eq? @_attr_name "lang")
+            (quoted_attribute_value
+                (attribute_value) @language
+            )
+        )
+    )
+    (raw_text) @content
+)
 
-; match scss
-((style_element
-  (start_tag
-    (attribute
-      (attribute_name) @_name
-      (quoted_attribute_value (attribute_value) @_value)))
-    (raw_text) @content)
-  (#eq? @_name "lang")
-  (#eq? @_value "scss")
-  (#set! "language" "scss"))
+; Match style tags without a lang attribute
+(style_element
+    (start_tag
+        (attribute
+            (attribute_name) @_attr_name
+        )*
+    )
+    (raw_text) @content
+    (#not-any-of? @_attr_name "lang")
+    (#set! language "css")
+)
 
-((raw_text_expr) @content
-  (#set! "language" "javascript"))
 
-((raw_text_each) @content
-  (#set! "language" "javascript"))
+; Downstream TODO: Style highlighting for `style:background="red"` and `style="background: red"` strings
```

Comment:
fyi these are outdated

Reply:
Good point - I just wonder whether it will even properly highlight since typescript might get confused and think it's a top level statement?

Reply:
Did you mean CSS? In either case, from my testing, it seems to work well. I took a glance at Tree-sitter CSS, and it allows top-level CSS assignments. And TypeScript, I think it's valid to put stuff like this on the top level of a TS file, so I see no reason why it should not work.

Reply:
@matubu anyways, I think this is potentially a good reason for why we should potentially extend some of the typescript tree sitter in order to parse the expressions in the ways we want to do (but probably in a separate PR as this is not a simple change, and will need to be proposed with the folks at `tree-sitter-grammars`

Anyhow, I will try to add TS syntax highlighting for the generics attribute

Reply:
<img width="267" alt="image" src="https://github.com/user-attachments/assets/9d1d876c-9910-4f89-bf47-9d9fcad1adb4">

@matubu unfortunately I was right, treating the `generics="T"` attribute as typescript creates the improper syntax highlighting

Reply:
<img width="678" alt="image" src="https://github.com/user-attachments/assets/74411583-a84c-41ea-a119-6b5674f678ef">

fyi ^

Reply:
> I think for now, it's best to keep it as a string and propose these more advanced changes inside of tree-sitter-svelte.

My proposed changed wouldn't fix the issue at it is only a change for svelte expression. Because the svelte parser has to "understand" the js/ts syntax, stuff like this: `{1+1}` `attribute={test}`.